schema/core.go

```diff
@@ -175,32 +175,54 @@ func (a CoreAttribute) validate(attribute interface{}) (interface{}, *errors.Sci
 		return nil, nil
 	}
 
-	if a.multiValued {
-		// return false if the multivalued attribute is not a slice.
-		arr, ok := attribute.([]interface{})
-		if !ok {
-			return nil, &errors.ScimErrorInvalidSyntax
+	if !a.multiValued {
+		return a.validateSingular(attribute)
+	}
+
+	switch arr := attribute.(type) {
+	case map[string]interface{}:
+		// return false if the multivalued attribute is empty.
+		if a.required && len(arr) == 0 {
+			return nil, &errors.ScimErrorInvalidValue
+		}
+
+		validMap := make(map[string]interface{}, len(arr))
+		for k, v := range arr {
+			for _, sub := range a.subAttributes {
+				if strings.EqualFold(sub.name, k) {
+					_, scimErr := sub.validate(v)
+					if scimErr != nil {
+						return nil, scimErr
+					}
+					validMap[sub.name] = v
+				}
+			}
+		}
+		if len(validMap) != len(arr) {
```

Comment:
Values that are not in the schema should be ignored. This means that the length of `validMap` is not necessarily as long as `arr`.

Reply:
Thanks for your review.
Fixed it. https://github.com/elimity-com/scim/pull/106/commits/dd195a40654cac4f0cf27c23951c0eb5d93139e3